src/lib/x509/x509_ext.h

```diff
@@ -568,6 +570,192 @@ class BOTAN_PUBLIC_API(3, 5) TNAuthList final : public Certificate_Extension {
       std::vector<Entry> m_tn_entries;
 };
 
+/**
+ * IP Address Blocks Extension
+ *
+ * RFC 3779 X.509 Extensions for IP Addr
+ *
+*/
+class BOTAN_PUBLIC_API(3, 6) IPAddressBlocks final : public Certificate_Extension {
+   public:
+      enum class BOTAN_PUBLIC_API(3, 6) Version : uint8_t {
+         IPv4 = 4,
+         IPv6 = 16,
+      };
+
+      template <Version V>
+      class BOTAN_PUBLIC_API(3, 6) IPAddress final {
+         public:
+            void set_from_bytes(const std::vector<uint8_t>& v);
+
+            std::array<uint8_t, static_cast<size_t>(V)> value() const { return m_value; }
+
+         private:
+            std::array<uint8_t, static_cast<size_t>(V)> m_value;
+      };
+
+      template <Version V>
+      class BOTAN_PUBLIC_API(3, 6) IPAddressOrRange final : public ASN1_Object {
+         public:
+            void encode_into(DER_Encoder&) const override;
+            void decode_from(class BER_Decoder& from) override;
+
+            IPAddress<V> min() const { return m_min; }
+
+            IPAddress<V> max() const { return m_max; }
+
+            void set_min(const IPAddress<V>& min) { m_min = min; }
+
+            void set_max(const IPAddress<V>& max) { m_max = max; }
```

Comment:
General comment: I'd prefer if possible (and it seems like it is from my current understanding) that values are set once in a constructor and we don't have any setters.